dbusmock/templates/bluez5.py

```diff
@@ -652,3 +699,69 @@ def DisconnectDevice(_self, adapter_device_name, device_address):
             [],
         ],
     )
+
+
+def RegisterAdvertisement(manager, adv_path, options):  # pylint: disable=unused-argument
+    if adv_path in manager.advertisements:
+        raise dbus.exceptions.DBusException("Already registered: " + adv_path, name="org.bluez.Error.AlreadyExists")
+
+    if len(manager.advertisements) >= manager.max_advertisement_instances:
+        raise dbus.exceptions.DBusException(
+            f"Maximum number of advertisements reached: {manager.max_advertisement_instances}",
+            name="org.bluez.Error.NotPermitted",
+        )
+
+    manager.advertisements.append(adv_path)
+
+    manager.UpdateProperties(
+        LE_ADVERTISING_MANAGER_IFACE,
+        {
+            "ActiveInstances": dbus.Byte(len(manager.advertisements)),
+            "SupportedInstances": dbus.Byte(manager.max_advertisement_instances - len(manager.advertisements)),
+        },
+    )
+
+
+def UnregisterAdvertisement(manager, adv_path):
+    try:
+        manager.advertisements.remove(adv_path)
+    except ValueError:
+        raise dbus.exceptions.DBusException(
+            "Unknown advertisement: " + adv_path, name="org.bluez.Error.DoesNotExist"
+        ) from None
+
+    manager.UpdateProperties(
+        LE_ADVERTISING_MANAGER_IFACE,
+        {
+            "ActiveInstances": dbus.Byte(len(manager.advertisements)),
+            "SupportedInstances": dbus.Byte(manager.max_advertisement_instances - len(manager.advertisements)),
+        },
+    )
+
+
+@dbus.service.method(BLUEZ_MOCK_IFACE, in_signature="s", out_signature="s")
+def AddAdvertisement(self, adv_name):
+    """Convenience method to add an Advertisement object
+
+    Returns the new object path.
+    """
+    path = "/org/dbusmock/bluez/advertisement/" + adv_name
+
+    adv_properties = {
+        "Type": dbus.String("broadcast", variant_level=1),
+        "ManufacturerData": dbus.Dictionary(
+            {dbus.Int16(0x0397): dbus.Array([0x00, 0x01], variant_level=2)}, signature="qv", variant_level=1
```

Comment:
Again, magic numbers -- this could use a comment where these come from, or what they mean?

Reply:
Have added some comments and switched to the reserved company identifier `0xFFFF` which is specifically for testing.